net/ip.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+/**
+ * Validates whether a given string is a valid IPv4 address
+ *
+ * @param addr IPv4 address in a string format (e.g., "192.168.0.1").
+ * @returns A boolean indicating if the string is a valid IPv4 address
+ *
+ * @example Check if the address is a IPv4
+ * ```ts no-assert ignore
+ * import { isIPv4 } from "@std/net/ip"
+ * import { assert, assertFalse } from "@std/assert"
+ *
+ * const correctIp = "192.168.0.1"
+ * const incorrectIp = "192.168.0.256"
+ *
+ * assert(isIPv4(correctIp))
+ * assertFalse(isIPv4(incorrectIp))
+ *
+ * ```
+ */
+export function isIPv4(addr: string): boolean {
+  const parts = addr.split(".");
+
+  if (parts.length <= 0 || parts.length < 4) return false;
+
+  for (const part of parts) {
+    const n = parseInt(part);
```

Comment:
because `parseInt` returns `1` for `"1a"` for example, `isIPv4("1a.1.1.1")` is currently true